code/obj/item/mops_cleaners.dm

```diff
@@ -557,11 +557,14 @@ TRASH BAG
 
 /obj/item/sponge/process()
 	if (!src.reagents) return
-	if (!istype(src.loc,/turf/simulated/floor)) return
+	if (!istype(src.loc,/turf/simulated/floor) && !istype(src.loc,/turf/space/fluid) ) return
 	if (src.reagents && src.reagents.total_volume >= src.reagents.maximum_volume) return
 	var/turf/simulated/floor/T = src.loc
 	if (T.active_liquid && T.active_liquid.group)
 		T.active_liquid.group.drain(T.active_liquid,1,src)
+	var/turf/space/fluid/F = src.loc
+	if(istype(F))
+		src.reagents.add_reagent(ocean_reagent_id,10)
```

Comment:
* Kind of weird that these didnt allow unsimulated floors so added that...
* no need to store the fluid turf in a var since we aren't getting the ocean reagent from it but instead of going to the global var
* Existing code had a redundant check of reagents
```suggestion
	if (src.reagents.total_volume >= src.reagents.maximum_volume) return
	if (isfloor(src.loc))
		var/turf/T = src.loc
		if (T.active_liquid && T.active_liquid.group)
			T.active_liquid.group.drain(T.active_liquid,1,src)
	else if (istype(src.loc, turf/space/fluid))
		src.reagents.add_reagent(ocean_reagent_id,10)
```